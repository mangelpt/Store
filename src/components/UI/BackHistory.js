import React from 'react'
import styled from '@emotion/styled';

const StyledDiv = styled.div`
width: 75vw;
display: flex;
height: fit-content;
font-size: 20px;
font-weight: 600;
margin:10px auto;
padding: 10px;
gap: 70px;
button{
  border: none;
  background-color: transparent;
  cursor: pointer;
}
svg{
  width: 20px;
  color: #000;
}
`;
export const BackHistory = ({ justify, children }) => {
  return (
    <StyledDiv style={{ justifyContent: `${justify}` }} >
      {children}
    </StyledDiv>
  )
}
