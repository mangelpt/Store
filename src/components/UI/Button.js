import styled from "@emotion/styled"

const Bttn = styled.button`
  background: #FA4A0C;
  border-radius: 30px;
  padding: 12px 16px;
  height: 70px;
  width: 310px;
  font-weight: 600;
  font-size: 18px;
  line-height: 23px;
  color: #FFFFFF;
  text-align: center;
  border: none;
  cursor: pointer;
  
  ${props => props.prefix==="foot" && `
    position: fixed;
    bottom: 104px;
  `} 

  &:focus {
    outline: none;
  }
`

export function Button(props) {
  return (
    <Bttn onClick={props.fnc} prefix={props.prefix}>{props.text}</Bttn>
  )
}
