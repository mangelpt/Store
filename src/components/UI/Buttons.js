import styled from "@emotion/styled";

export const ButtonWhite = styled.button`
  padding: 15px 15px;
  font-size: 18px;
  font-weight: bold;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: orange;
  border: none;
  border-radius: 20px;
  cursor: pointer;
  margin: auto;
  width: 75vw;
  color: #000;
  svg {
    width: 25px;
  }
`;
