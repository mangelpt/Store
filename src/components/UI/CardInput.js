import styled from "@emotion/styled"
import { Lbl } from "./Labels"
import  {Input} from "./Input"

const ContainerInput = styled.div`
    display: flex;
    flex-direction: column;
    width:208px;
    height:50px;
`

export function CardInput({textlabel, name, type, placeholder}) {
  return (
    <>
    <ContainerInput>
      <Lbl>{textlabel}</Lbl>
      <Input type={type}
          name={name}
          placeholder={placeholder}
      >
      </Input>
    </ContainerInput>  
    </>
  )
}