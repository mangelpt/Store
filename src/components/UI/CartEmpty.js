import styled from "@emotion/styled";

export function CartEmpty() {

  const ContainerCartEmpty = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    svg {
      width: 100px;
      height: 100px;
    }
    .iconcart {
      fill: none;
      stroke: #C7C7C7;
    }
    h1 {
      font-family: "Source Sans Pro";
      font-style: normal;
      font-weight: 600;
      font-size: 28px;
      line-height: 35px;
      text-align: center;
      color: #333333;
    }
  `;

  return (
    <ContainerCartEmpty>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        class="h-6 w-6"
        fill="none"
        viewBox="0 0 24 24"
        stroke="currentColor"
      >
        <path
          class="iconcart"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
        />
      </svg>
      <h1>No order yet</h1>
    </ContainerCartEmpty>
  );
}
