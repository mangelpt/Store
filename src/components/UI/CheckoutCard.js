import styled from "@emotion/styled";

const Checkout = styled.div`
  background: #FFFFFF;
  box-shadow: 0px 10px 40px rgba(0, 0, 0, 0.03);
  border-radius: 20px;
  padding: 25px 53px 20px 30px;
  width: 232px;
  margin-bottom: 10px;
  form {
    width: 232px;
    display: flex;
    flex-direction: column;
  }

  input {
    font-weight: 400;
    font-size: 18px;
    line-height: 23px;
    color: #333333;
    border: none;
    background: none;
    outline: none;
  }

  input[name="name"] {
    font-weight: 600;
  }

  input::placeholder {
    opacity: 0.5;
  }

  div {
    margin: 7px 0 10px 0;
    opacity: 0.2;
    border-top: 0.5px solid #000000; 
  }
`;

export function CheckoutCard(props) {
  return (
    <Checkout>
      <form>
        <input
          type="text"
          name="name"
          value={props.name}
          placeholder="name"
          onChange={props.onChange}
          disabled={props.disabled}
        />
        <div />
        <input
          type="text"
          name="address"
          value={props.address}
          onChange={props.onChange}
          placeholder="address" 
          disabled={props.disabled}
        />
        <div />
        <input
          type="tel"
          name="phone"
          value={props.phone}
          placeholder="phone"
          onChange={props.onChange}
          disabled={props.disabled}
        />
      </form>
    </Checkout>
  )
}