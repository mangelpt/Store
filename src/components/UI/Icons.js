export const ArrowIcon = () => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M15 19l-7-7 7-7" />
    </svg>
  )
}

export const ArrowIconRight = () => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" className="h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
      <path fillRule="evenodd" d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z" clipRule="evenodd" />
    </svg>
  )
}

export const ArrowIconDown = () => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M19 9l-7 7-7-7" />
    </svg>
  )
}

export const ArrowIconUp = () => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M5 15l7-7 7 7" />
    </svg>
  )
}
export const HomeIcon = () => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6" />
    </svg>
  )
}

export const HistoryIcon = () => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
    </svg>
  )
}

export const UserIcon = () => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z" />
    </svg>
  )
}

export const CartIcon = () => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z" />
    </svg>
  )
}

export const SearchIcon = () => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z" />
    </svg>
  )
}

export const CalendarIcon = () => {
  return (
    <svg
      width="108"
      height="119"
      viewBox="0 0 108 119"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M95.4167 11.8333H89.5V0H77.6667V11.8333H30.3333V0H18.5V11.8333H12.5833C6.075 11.8333 0.75 17.1583 0.75 23.6667V106.5C0.75 113.008 6.075 118.333 12.5833 118.333H95.4167C101.925 118.333 107.25 113.008 107.25 106.5V23.6667C107.25 17.1583 101.925 11.8333 95.4167 11.8333ZM95.4167 106.5H12.5833V47.3333H95.4167V106.5ZM12.5833 35.5V23.6667H95.4167V35.5H12.5833ZM24.4167 59.1667H83.5833V71H24.4167V59.1667ZM24.4167 82.8333H65.8333V94.6667H24.4167V82.8333Z"
        fill="#C7C7C7"
      />
    </svg>
  );
}
