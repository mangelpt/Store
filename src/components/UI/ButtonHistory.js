import styled from '@emotion/styled';
import { Link } from "react-router-dom";
import { ArrowIconRight } from "./Icons";

const StyledDiv = styled.div`
.link{
  padding: 15px 15px;
  text-decoration: none;
  font-size: 18px;
  font-weight: bold;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #fff;
  border: none;
  border-radius: 20px;
  cursor: pointer;
  margin: auto;
  width: 75vw;
  color: #000;
  svg {
    width: 25px;
  }
}

`;
export const ButtonHistory = () => {
  return (
    <StyledDiv>
      <Link to="/history" className="link">
        History
        <ArrowIconRight />
      </Link>
    </StyledDiv>
  )
}

