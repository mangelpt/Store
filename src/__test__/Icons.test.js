import React from "react";
import renderer from "react-test-renderer";
import { ArrowIcon, ArrowIconDown, ArrowIconRight, ArrowIconUp, CalendarIcon, CartIcon, CartIconBar, HistoryIcon, HomeIcon, SearchIcon, SearchIconBar, UserIcon } from "../components/UI/Icons";

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ArrowIcon/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ArrowIconRight/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ArrowIconDown/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ArrowIconUp/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<HomeIcon/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<HistoryIcon/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<UserIcon/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<CartIcon/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SearchIcon/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<CalendarIcon/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SearchIconBar/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Styled Icons", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<CartIconBar/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

