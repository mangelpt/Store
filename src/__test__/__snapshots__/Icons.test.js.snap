// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Icons renders correctly 1`] = `
<svg
  className="h-6 w-6"
  fill="none"
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M15 19l-7-7 7-7"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Styled Icons renders correctly 2`] = `
<svg
  className="h-5 w-5"
  fill="currentColor"
  viewBox="0 0 20 20"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clipRule="evenodd"
    d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
    fillRule="evenodd"
  />
</svg>
`;

exports[`Styled Icons renders correctly 3`] = `
<svg
  className="h-6 w-6"
  fill="none"
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M19 9l-7 7-7-7"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Styled Icons renders correctly 4`] = `
<svg
  className="h-6 w-6"
  fill="none"
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M5 15l7-7 7 7"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Styled Icons renders correctly 5`] = `
<svg
  className="h-6 w-6"
  fill="none"
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Styled Icons renders correctly 6`] = `
<svg
  className="h-6 w-6"
  fill="none"
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Styled Icons renders correctly 7`] = `
<svg
  className="h-6 w-6"
  fill="none"
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Styled Icons renders correctly 8`] = `
<svg
  className="h-6 w-6"
  fill="none"
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Styled Icons renders correctly 9`] = `
<svg
  className="h-6 w-6"
  fill="none"
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Styled Icons renders correctly 10`] = `
<svg
  fill="none"
  height="119"
  viewBox="0 0 108 119"
  width="108"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M95.4167 11.8333H89.5V0H77.6667V11.8333H30.3333V0H18.5V11.8333H12.5833C6.075 11.8333 0.75 17.1583 0.75 23.6667V106.5C0.75 113.008 6.075 118.333 12.5833 118.333H95.4167C101.925 118.333 107.25 113.008 107.25 106.5V23.6667C107.25 17.1583 101.925 11.8333 95.4167 11.8333ZM95.4167 106.5H12.5833V47.3333H95.4167V106.5ZM12.5833 35.5V23.6667H95.4167V35.5H12.5833ZM24.4167 59.1667H83.5833V71H24.4167V59.1667ZM24.4167 82.8333H65.8333V94.6667H24.4167V82.8333Z"
    fill="#C7C7C7"
  />
</svg>
`;

exports[`Styled Icons renders correctly 11`] = `
<svg
  fill="none"
  height="20"
  viewBox="0 0 20 20"
  width="20"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M19 19L14.65 14.65M17 9C17 13.4183 13.4183 17 9 17C4.58172 17 1 13.4183 1 9C1 4.58172 4.58172 1 9 1C13.4183 1 17 4.58172 17 9Z"
    stroke="black"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Styled Icons renders correctly 12`] = `
<svg
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    opacity="0.3"
  >
    <path
      d="M9 22C9.55228 22 10 21.5523 10 21C10 20.4477 9.55228 20 9 20C8.44772 20 8 20.4477 8 21C8 21.5523 8.44772 22 9 22Z"
      stroke="black"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
    <path
      d="M20 22C20.5523 22 21 21.5523 21 21C21 20.4477 20.5523 20 20 20C19.4477 20 19 20.4477 19 21C19 21.5523 19.4477 22 20 22Z"
      stroke="black"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
    <path
      d="M1 1H5L7.68 14.39C7.77144 14.8504 8.02191 15.264 8.38755 15.5583C8.75318 15.8526 9.2107 16.009 9.68 16H19.4C19.8693 16.009 20.3268 15.8526 20.6925 15.5583C21.0581 15.264 21.3086 14.8504 21.4 14.39L23 6H6"
      stroke="black"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
  </g>
</svg>
`;
